File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        """
        for i in range(len(nums)):
            other = target - nums[i]
            if other in nums and nums.index(other) != i:
                return list([i, nums.index(target - nums[i])])
        """
        
        seen= {}
        for i, num in enumerate(nums):
            complement= target - num
            if complement in seen:
                return([seen[complement], i])
            elif complement not in seen:
                seen[num] = i

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        int[] result = new int[2];
        Map<Integer, Integer> seen = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            for (Map.Entry<Integer, Integer> m : seen.entrySet()) {
                if (m.getKey() == complement) {
                    result[0] = m.getValue();
                    result[1] = i;
                    return result;
                }
            }
            seen.put(nums[i], i);
        }
        return result;
    }
}
